src/main/java/API_hw2/core/TrelloServiceObj.java

```diff
@@ -0,0 +1,187 @@
+import API_hw2.constants.BoardConstant;
+import API_hw2.constants.ResponseStatus;
+import API_hw2.constants.URLConstant;
+import com.google.gson.Gson;
+import com.google.gson.reflect.TypeToken;
+import API_hw2.constants.*;
+import io.restassured.RestAssured;
+import io.restassured.builder.RequestSpecBuilder;
+import io.restassured.builder.ResponseSpecBuilder;
+import io.restassured.http.ContentType;
+import io.restassured.http.Method;
+import io.restassured.response.Response;
+import io.restassured.specification.RequestSpecification;
+import io.restassured.specification.ResponseSpecification;
+import org.apache.http.HttpStatus;
+
+import java.net.URI;
+import java.nio.charset.Charset;
+import java.util.HashMap;
+import java.util.Map;
+
+import static io.restassured.http.ContentType.*;
+import static org.hamcrest.Matchers.*;
+
+public class TrelloServiceObj {
+
+    public static String URL;
+    private Method requestMethod;
+
+    private Map<String, String> parameters;
+
+    private TrelloServiceObj(Map<String, String> parameters, Method method) {
+        this.parameters = parameters;
+        this.requestMethod = method;
+    }
+
+    public static ApiRequestBuilder requestBuilder() {
+        return new ApiRequestBuilder();
+    }
+
+    public static class ApiRequestBuilder {
+        private Map<String, String> parameters = new HashMap<>();
+        private Method requestMethod = Method.GET;
+
+        public ApiRequestBuilder setMethod (Method method){
+            requestMethod = method;
+            return this;
+        }
+
+        public ApiRequestBuilder id(String id) {
+            parameters.put(BoardConstant.ID.getConstantName(), id);
+            return this;
+        }
+
+        public ApiRequestBuilder name(String name) {
+            parameters.put(BoardConstant.NAME.getConstantName(), name);
+            return this;
+        }
+
+        public ApiRequestBuilder closed(Boolean closingState) {
+            parameters.put(BoardConstant.CLOSED.getConstantName(), closingState.toString());
+            return this;
+        }
+
+        public ApiRequestBuilder desc(String desc) {
+            parameters.put(BoardConstant.DESC.getConstantName(), desc);
+            return this;
+        }
+
+        public ApiRequestBuilder customParameter(String paramName, String param) {
+            parameters.put(paramName, param);
+            return this;
+        }
+
+        public TrelloServiceObj buildRequest() {
+            return new TrelloServiceObj(parameters, requestMethod);
+        }
+    }
+
+    private static RequestSpecification baseRequestConfiguration() {
+        return new RequestSpecBuilder()
+                //.setAccept(ContentType.JSON)
+                //.addQueryParam("requestNumber", ++requestNumber)
+                .setRelaxedHTTPSValidation()
+                .setBaseUri(URI.create(URLConstant.BASE.get()))
+                .addParam("key","05d638746f0f484a63b83713772fac99")
```

Comment:
Don't keep the credentials hard coded. Set environment variables or (for educational purposes) read the data from property file.

Reply:
added properties file